### ISSUE TITLE
Plugin reporting nonexistent error: "expected `&A`, found `A`"

### ISSUE THREAD

**Issue by michaelkourlas** [intellij-rust/intellij-rust#2528]
The plugin is reporting that the following code contains an error even though it compiles successfully:

```rust
use std::fmt::Display;
use std::fmt::Formatter;
use std::fmt::Error;

pub enum A {
    B
}

impl Display for A {
    fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
        match self {
            A::B => Ok(())
        }
    }
}
```

The reported error is on the line `A::B => Ok(())`. It reads:

```
mismatched types [E0308] 
expected `&A`, found `A`
```

Changing `self` to `*self` fixes it, but from what I can tell this shouldn't be necessary.

Using rust 1.26.0 and intellij-rust 0.2.0.2094-181.

**Comment by Undin** [comment]
Duplicate of #2457
